### PR TITLE
change compiler warning flags to reflect minimum requirement of C++14

### PR DESCRIPTION
**Pull request by FranciscoPombal**

### CODE REVIEW

**Comment by arvidn** [comment]
looks good, would you mind applying the same change to the `Jamfile` while you're at it?

**Comment by FranciscoPombal** [comment]
@arvidn done.

**Comment by arvidn** [comment]
It seems you need to leave the `no-c++98-compat` warning flags in. https://travis-ci.org/github/arvidn/libtorrent/jobs/701215876#L334

File: CMakeLists.txt [hunk 1]
```diff
@@ -471,11 +471,11 @@ if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
 		-Weverything
 		-Wno-documentation
 		-Wno-c++98-compat-pedantic
+		-Wno-c++11-compat-pedantic
 		-Wno-padded
 		-Wno-global-constructors
 		-Wno-exit-time-destructors
 		-Wno-weak-vtables
-		-Wno-return-std-move-in-c++11
```
Comment (arvidn) [inline CMakeLists.txt:478]: it seem you need to leave this one here a well: https://travis-ci.org/github/arvidn/libtorrent/jobs/701265066#L1407

Comment (FranciscoPombal) [reply CMakeLists.txt]: fixed

**Comment by FranciscoPombal** [comment]
@arvidn looks like all of the CI checks have passed this time. For some reason I can't see the link to Travis on this page, it's this one: https://travis-ci.org/github/arvidn/libtorrent/builds/701313069

**Pull request merged**